openml_OS/models/api/v1/Api_data.php

```diff
@@ -294,11 +294,27 @@ private function data_edit() {
       $this->returnError( 1064, $this->version );
       return;
     }
-    // check if user owns dataset
+
+    // create a copy of the dataset with same file but different ID and different user ID
```

Comment:
I would strongly agree against doing this! By doing this, you change the semantic meaning of an "edit" function, making the system unpredictable and hard to maintain